gen_build: Avoid endless configure loop

The build.ninja.d dependency file written by bin/gen_build adds "." as
a dependency of build.ninja. This causes the generation rule for the
build.ninja file to re-run endlessly because the mtime of the directory changes
every time the build.ninja file is written.

Remove all dependencies instead, which fixes the build for us, and is not
a problem because we don't update the sources in-place and rebuild.

Upstream-Status: Inappropriate [Configuration]
diff --git a/bin/gen_build b/bin/gen_build
index b3412da4..29ce2f0c 100755
--- ./bin/gen_build
+++ ./bin/gen_build
@@ -562,10 +562,7 @@ Dir.chdir(File.dirname(path)) do
       builder.write(io, Target.new(path))
 
       atomic_write("#{builddir}/build.ninja.d") do |io|
-        dep = BuildDependencies.to_a.map do |file|
-          file.gsub(%r{^\./|/\.(?=/|$)}, '')
-        end
-        builder.dependencies(io, dep)
+        io << "build.ninja: \n"
       end
     end
   end
